pkg/models/models.go

```diff
@@ -63,10 +63,27 @@ func (manager Manager) String() string {
 	return string(manager)
 }
 
+// IDFromString converts a string to an ID, typically provided from a user.
+// The OpenAPI spec defines precisely the regex that UUIDs must conform to:
+//
+//	^[0-9a-fA-F]{8}\\-[0-9a-fA-F]{4}\\-4[0-9a-fA-F]{3}\\-[8-b][0-9a-fA-F]{3}\\-[0-9a-fA-F]{12}$
+//
+// This is the strictest possible interpretation for a "V4 UUID RFC 4122 Variant" (See RFC 4122, section-4.4) and means we must refuse
+// even UUIDs that have the correct format but are not technically V4 UUID using the RFC 4122 variant.
+//
+// See https://github.com/astm-utm/Protocol/blob/cb7cf962d3a0c01b5ab12502f5f54789624977bf/utm.yaml#L128 for more details
 func IDFromString(s string) (ID, error) {
 	id, err := uuid.Parse(s)
 	if err != nil {
-		return ID(""), stacktrace.Propagate(err, "Error parsing ID in UUID format")
+		return ID(""), stacktrace.Propagate(err, "Error parsing ID in UUID V4 format: `%s`", s)
+	}
+	// TODO check the other required fields: the UUID lib reports the value of the version field
+	// but does not validate that the other V4 specific bits are correct.
+	if id.Version() != 4 {
+		return ID(""), stacktrace.NewError("UUID must be a V4 version, was: `%v`, id: `%s`", id.Version(), s)
+	}
+	if id.Variant() != uuid.RFC4122 {
```

Comment:
nit: Version only makes sense if the Variant is RFC4122, right?  If so, this check should probably be performed first.

Reply:
Agreed.  In RFC they note that "Variant" is really a "Type" and they discuss it prior to version for that reason... so if it isn't RFC4122, then basically bets are off in terms of the rest of the formatting.